crates/services/txpool_v2/src/collision_manager/mod.rs

```diff
@@ -17,57 +17,21 @@ use crate::error::Error;
 
 pub mod basic;
 
-/// The reason why a transaction collides with another.
-/// It also contains additional information about the collision.
-#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
-pub enum CollisionReason {
```

Comment:
Yes I was planning this to do it in the last PR that would clean up the error type because right now I used the one from before with some mess and I think it should be changed but I can do it a bit in this PR.

Reply:
I have added specifications for the `CollisionReason` for the error and the rest of the refactor error type will be in https://github.com/FuelLabs/fuel-core/pull/2216